init.c

```diff
@@ -99,44 +99,46 @@ struct MyVar
 {
   char *name;
   char *value;
-  struct MyVar *next;
+  TAILQ_ENTRY(MyVar) entries;
 };
 
-static struct MyVar *MyVars;
+TAILQ_HEAD(MyVarList, MyVar);
+static struct MyVarList MyVars = TAILQ_HEAD_INITIALIZER(MyVars);
 
 void myvar_set(const char *var, const char *val)
 {
-  struct MyVar **cur = NULL;
-
-  for (cur = &MyVars; *cur; cur = &((*cur)->next))
-    if (mutt_str_strcmp((*cur)->name, var) == 0)
-      break;
-
-  if (!*cur)
-    *cur = mutt_mem_calloc(1, sizeof(struct MyVar));
+  struct MyVar *myv = NULL;
+  struct MyVar *new = NULL;
```

Comment:
How about doing without the second variable and just reusing `myv` later one?

Reply:
This was the previous version of this code. I've introduced another variable just for clarity. No other reasons.

Reply:
Removed.